crates/ecs/src/lib.rs

```diff
@@ -503,111 +528,376 @@ pub struct World {
     /// By taking the intersection of these vectors you will know which `Archetype` contain both A and B.
     /// This could be the archetypes: (A,B), (A,B,C), (A,B,...) etc.
     component_typeid_to_archetype_indices: HashMap<TypeId, HashSet<ArchetypeIndex>>,
-    /// todo(#72): Contains all `TypeId`s of the components that `World` stores.
-    /// todo(#72): Remove, only used to showcase how archetypes can be used in querying.
-    stored_types: Vec<TypeId>,
 }
 
 type ReadComponentVec<'a, ComponentType> = Option<RwLockReadGuard<'a, Vec<Option<ComponentType>>>>;
 type WriteComponentVec<'a, ComponentType> =
     Option<RwLockWriteGuard<'a, Vec<Option<ComponentType>>>>;
+type TargetArchSourceTargetIDs = (Option<ArchetypeIndex>, Vec<TypeId>, Vec<TypeId>);
+
+enum ArchetypeMutation {
+    Removal(Entity),
+    Addition(Entity),
+}
 
 impl World {
-    /// todo(#72): Adds the Component to the entity by storing it in the `Big Archetype`.
-    /// todo(#72): Adds a new component vec to `Big Archetype` if it does not already exist.
-    fn create_component_vec_and_add<ComponentType: Debug + Send + Sync + 'static>(
+    /// Returns the archetype index of the archetype that contains all
+    /// components types that the entity is tied to -+ the generic type
+    /// supplied to the function given "removal flag" status.
+    /// None is returned if no archetype containing only the sought after component types exist.
+    /// The type Ids contained within the archetype the entity is existing in and the type ids for
+    /// sought archetype are also returned.
+    ///
+    /// Set removal to true or false depending on if the supplied generic type should be added to or
+    /// removed from the specifie entity.
+    /// For example call world.find_target_archetype::<u32>(entity, true)
+    /// to fetch the archetype index of the archetype
+    /// containing all component types except u32 that the entity is tied to.
+    ///
+    /// world.find_target_archetype::<u32>(entity, false) would return the archetype index of the
+    /// archetype containing all component types that the entity is tied to with the addition
+    /// of u32.
+    fn find_target_archetype<ComponentType: Debug + Send + Sync + 'static>(
```

Comment:
Very nice, thanks! 

You should make your code examples into [doctests](https://doc.rust-lang.org/rustdoc/write-documentation/documentation-tests.html) so we'll get warned if they become out-of-date!

```suggestion
    /// Returns the archetype index of the archetype that contains all
    /// components types that the entity is tied to -+ the generic type
    /// supplied to the function given "removal flag" status.
    /// `None` is returned if no archetype containing only the sought after component types exist.
    /// The type ids contained within the archetype the entity is existing in and the type ids for
    /// sought archetype are also returned.
    ///
    /// Set removal to `true` or `false` depending on if the supplied generic type should be added to or
    /// removed from the specified entity.
    /// For example call world.find_target_archetype::<u32>(entity, true)
    /// to fetch the archetype index of the archetype
    /// containing all component types except u32 that the entity is tied to.
    ///
    /// world.find_target_archetype::<u32>(entity, false) would return the archetype index of the
    /// archetype containing all component types that the entity is tied to with the addition
    /// of u32.
    fn find_target_archetype<ComponentType: Debug + Send + Sync + 'static>(
```